File: FrontEnd/lib/domain/interfaces/usecases/auth/i_validate_token_use_case.dart
/// @file i_validate_token_use_case.dart
/// @description Interfaz del caso de uso de validación de token JWT.
/// @module Core
/// @layer Domain
library;

import '../../../entities/user.dart';

/// Contrato del caso de uso de validación de sesión.
abstract interface class IValidateTokenUseCase {
  /// Valida el token almacenado y devuelve el usuario autenticado.
  ///
  /// [throws] AppError.unauthorized — token inválido o expirado.
  Future<User> execute();
}

File: BackEnd/src/domain/interfaces/usecases/auth/IValidateTokenUseCase.ts
/**
 * @file IValidateTokenUseCase.ts
 * @description Interfaz del caso de uso de validación de token JWT.
 * @module Auth
 * @layer Domain
 */

import type { UserResponseDTO } from '../../../dtos/user/user-response.dto.js';

/** Contrato del use case de validación de token. */
export interface IValidateTokenUseCase {
  /**
   * @param token — Token JWT a validar.
   * @returns Datos públicos del usuario si el token es válido.
   * @throws {UnauthorizedException} Si el token es inválido o ha expirado.
   */
  execute(token: string): Promise<UserResponseDTO>;
}
